src/main/java/com/ericsson/gerrit/plugins/eiffel/state/SourceChangeCreatedState.java

```diff
@@ -0,0 +1,50 @@
+/*
+   Copyright 2019 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.ericsson.gerrit.plugins.eiffel.state;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.ConnectException;
+import java.sql.SQLException;
+
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeCreatedEvent;
+import com.ericsson.gerrit.plugins.eiffel.handlers.NoSuchElementException;
+import com.ericsson.gerrit.plugins.eiffel.handlers.Table;
+
+public class SourceChangeCreatedState extends State {
+
+    public SourceChangeCreatedState(File pluginDir) {
+        super(pluginDir);
+    }
+
+    @Override
+    public String getEventId(String project, String changeId) throws NoSuchElementException, ConnectException, FileNotFoundException {
+        return getLastSubmittedEiffelEvent(project, changeId, Table.SCC_TABLE);
+    }
+
+    @Override
+    public void setState(String eiffelEventId, EiffelEvent eiffelEvent)
```

Comment:
You still save the ID, not the entire event, this is also named eiffelEvent in DatabaseHandler, but should be ID, or actually maybe value as it could be something else in the future.